### PR TITLE
Add input/target Range Check for BCEWithLogitsLoss

### LINKED ISSUE #110865: BCEWithLogitsLoss: Check if labels within zero and one
### 🚀 The feature, motivation and pitch

`BCEWithLogitsLoss` currently performs no kind of checking wrt. the targets whatsoever. This means, that calling the loss function with targets not contained within the interval [0,1] succesfully computes values without throwing an error. However, these values don't have any kind of meaning.

In practice, though, passing invalid targets to `BCEWithLogitsLoss` can easily happen and can go unnoticed! Such mistakes can render whole experiments invalid; **if** they are even noticed in the first place!

For instance, this can happen if multi-class labels and predictions (i.e. 0,1,2,...,n) are erroneously used with BCEWithLogitsLoss. *Shouldn't this result in a shape-mismatch due to the prediction being a categorical distribution?* No, both predictions and targets will be broadcastable to each other in that case. In fact, commonly, their shape will be identical.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by isdanni**
Fixes #110865

### CODE REVIEW

File: aten/src/ATen/native/Loss.cpp [hunk 1]
```diff
@@ -360,6 +360,26 @@ Tensor binary_cross_entropy_with_logits(const Tensor& input, const Tensor& targe
   const Tensor& pos_weight = *pos_weight_maybe_owned;
 
     Tensor loss;
+
+    auto iter = TensorIteratorConfig()
+      .add_input(input)
+      .add_input(target)
+      .build();
+    
+      at::native::cpu_serial_kernel(
```
Comment (albanD) [inline aten/src/ATen/native/Loss.cpp:369]: Doing a full pass just to check error is not good for performance.

**Review by albanD — COMMENTED** [review]
cc @drisspg the cpu check is a good example for a debug mode
